### ISSUE TITLE
DataTable: when editable (cell), draggableRows and selection (single) -> PF(var).showCellEditor() does not work

### ISSUE THREAD

**Issue by quetsalkoatl** [primefaces/primefaces#2456]
## 1) Environment
- PrimeFaces version: 6.1
- Application server + version: Apache Tomcat 8.0.23
- Affected browsers: FF ESR 52.1.0, IE 11

## 2) Expected behavior
Single cell editor should appear on contextMenu action. `<p:contextMenu for="tblId">` and  `<p:menuitem value="Edit" icon="fa fa-pencil" onclick="PF('tblVar').showCellEditor();return false;"/>`

## 3) Actual behavior
On left-click the editor appears as expected, but the contextMenu action throws a Javascript error: TypeError: a is undefined - PrimeFaces.widget.DataTable<.showCellEditor - https://relmanplan-t/javax.faces.resource/components.js.xhtml:12:41442

## 4) Steps to reproduce
Create dataTable which is editable (editMode=cell), has draggable rows and selectionMode="single".

## 5) Sample XHTML
```xhtml
<p:dataTable id="tblId" var="row" value="${detail.data}" editable="true" editMode="cell" rowKey="#{row.order}"
             draggableRows="true" widgetVar="tblVar" selectionMode="single" selection="#{detail.selected}">
    <f:facet name="header">Excel Export</f:facet>
    <p:ajax event="rowReorder" listener="${detail.onReorder}" update="msg_ajax" />
    <p:ajax event="cellEdit" listener="${detail.onCell}" update="msg_ajax" />
    [colums]
    <p:column headerText="Datum" style="width: 70px; background-color: ${row.dateRGB}">
        <p:cellEditor>
            <f:facet name="output">
                <h:outputText value="${row.date}">
                    <f:convertDateTime pattern="dd.MM.yyyy"/>
                </h:outputText>
            </f:facet>
            <f:facet name="input">
                <p:calendar value="#{row.date}" pattern="dd.MM.yyyy" locale="de" navigator="true"
                            showOtherMonths="true" selectOtherMonths="true" placeholder="Datum" />
            </f:facet>
        </p:cellEditor>
    </p:column>
    [columns]
</p:dataTable>

<p:contextMenu for="tblId">
    <p:menuitem value="Edit" icon="fa fa-pencil" onclick="PF('tblVar').showCellEditor();return false;"/>
</p:contextMenu>
```

**Comment by jxmai** [comment]
@tandraschko  I don't think this issue is valid. `showCellEditor` does what it was intended to do. 

Please take a look,

https://github.com/primefaces/primefaces/blob/481fdff1c23dfff61425afd2800d3d93d4c2c5c3/src/main/resources/META-INF/resources/primefaces/datatable/datatable.js#L2351-L2376

The reporter tried to pass a null into the function above.

![snapshot1](https://user-images.githubusercontent.com/10467831/34086133-365d559a-e35f-11e7-9edb-05df2b13efb6.png)

Normally on left click we should expect to see this:

![snapshot1](https://user-images.githubusercontent.com/10467831/34086147-562f800a-e35f-11e7-806a-d0a30de83bea.png)


Basically the function call below is invalid in the first place. How can the showFunction knows which cell to show without getting a cell as an argument ?

`onclick="PF('tblVar').showCellEditor();return false;"`
